src/mkdir.js

```diff
@@ -83,16 +67,23 @@ function _mkdir(options, dirs) {
         fs.mkdirSync(dir, parseInt('0777', 8));
       }
     } catch (e) {
-      var reason;
-      if (e.code === 'EACCES') {
-        reason = 'Permission denied';
-      } else if (e.code === 'ENOTDIR' || e.code === 'ENOENT') {
-        reason = 'Not a directory';
-      } else {
-        /* istanbul ignore next */
-        throw e;
-      }
-      common.error('cannot create directory ' + dir + ': ' + reason, { continue: true });
+      // See mkdir(2) for details on each error case
+      // http://man7.org/linux/man-pages/man2/mkdir.2.html
+      var codeToMessage = {
+        EACCES: 'Permission denied',
+        EEXIST: 'File exists',
+        ENAMETOOLONG: 'File name too long',
+        ENOENT: 'No such file or directory',
+        ENOTDIR: 'Not a directory',
```

Comment:
I limited this to the list of error codes I could repro locally. I think that's fine for now, and we can add more later.

Reply:
Is this enum something that should be declared globally? Perhaps in `src/common.js`?

Reply:
Some of the error codes can be shared (e.g., `EEXIST`). However, there's no real advantage to moving these to common.js. It's the difference between `common.codes.EEXIST` and `'EEXIST'`.

When it comes to user-facing error messages, I think each command wants a different message as it sees fit.

Reply:
Ok, makes sense. This enum could at least be declared globally in this file.

Reply:
Done